store/src/main/java/discord4j/store/service/StoreServiceLoader.java

```diff
@@ -35,59 +34,58 @@
      *
-     * @return The best {@link StoreService} implementation.
+     * @param priorityOverrides Allows for manual overriding of {@link StoreService#priority()}.
      */
-    public StoreService getStoreService() {
-        return generalService;
-    }
+    @SuppressWarnings({"ComparatorCombinators", "Convert2Lambda"})
+    public StoreServiceLoader(Map<Class<? extends StoreService>, Integer> priorityOverrides) {
+        generalService = new Lazy<>(() -> {
+            List<StoreService> services = new ArrayList<>();
+            ServiceLoader<StoreService> serviceLoader = ServiceLoader.load(StoreService.class);
+
+            serviceLoader.iterator().forEachRemaining(services::add);
+
+            services.add(new NoOpStoreService()); //No-op does not use discovery since it is always present
+
+            services.sort(new Comparator<StoreService>() {
+                @Override
+                public int compare(StoreService ss1, StoreService ss2) {
+                    return Integer.compare(priorityOverrides.getOrDefault(ss1.getClass(), ss1.priority()),
+                            priorityOverrides.getOrDefault(ss2.getClass(), ss2.priority()));
+                }
+            });
+
+            StoreService generic = services.stream().filter(StoreService::hasGenericStores).findFirst().get();
+
+            StoreService primitive = services.stream().filter(StoreService::hasLongObjStores).findFirst().get();
+            if (primitive instanceof NoOpStoreService) { //Fallback to boxed impl if one is present
+                if (!(generic instanceof NoOpStoreService)) {
+                    return new ForwardingStoreService(generic);
+                }
+            }
 
-    /**
-     * Gets the service which will be used to provide generic stores.
-     *
-     * @return The generic store providing service.
-     */
-    StoreService getGenericStoreProvider() {
-        return services.stream().filter(StoreService::hasGenericStores).findFirst().get();
+            return generic == primitive ? generic : new ComposedStoreService(generic, primitive);
+        });
     }
 
     /**
-     * Gets the service which will be used to provide long-object stores.
+     * Gets the {@link StoreService} implementation to use.
      *
-     * @return The long-object store providing service.
+     * @return The best {@link StoreService} implementation.
      */
-    StoreService getLongObjStoreProvider() {
-        StoreService service = services.stream().filter(StoreService::hasLongObjStores).findFirst().get();
-
-        if (service instanceof NoOpStoreService) { //Fallback to boxed impl if one is present
-            StoreService backup = getGenericStoreProvider();
-            if (!(backup instanceof NoOpStoreService)) {
-                return new ForwardingStoreService(backup);
-            }
-        }
-
-        return service;
+    public StoreService getStoreService() {
```

Comment:
If we're exposing this class at the Client builder level, then this should be the only public method in this class, to clarify that one StoreService is the only thing needed from it for all the different Stores used.